### DISCUSSION TITLE
Contributing: How you test it?

### DISCUSSION THREAD

**Discussion by NicolasLopes7** [hadynz/obsidian-kindle-plugin#149] [Q&A]
Hey, I want to contribute to your project but I'm not sure about how to test it locally, which steps, .env are required...
can you create a `CONTRIBUTING.md` file or add these instructions to the README?

**Comment by hadynz** [comment]
That's a fair call. I've slowly started adding more unit test coverage as well to encourage more contributions.

Do you have examples of really good `CONTRIBUTING.md` examples that I can take into consideration as I do this?

**Comment by baipai** [comment]
Hi, I want to contribute to this project for the design part, what is the best way to be involved as well? @hadynz 

https://opensourcedesign.net/jobs/jobs/2022-07-16-design-ui-for-obsidians-kindle-plugin

**Reply by hadynz** [reply]
@baipai do you have a Kindle and use this plugin? If so, you'd be the perfect person to pair up with as you have all the required context.

**Reply by baipai** [reply]
@hadynz Yeah, I have kindle and play around with the plugin. I found out if I don't put the highlight but just put the note then the note won't show up. But maybe this is not the ideal use case scenario? I also checked amazon's own reading highlight and note system, it can show up there.

What's the best way to start with? what's the biggest problem now?

**Reply by hadynz** [reply]
> I found out if I don't put the highlight but just put the note then the note won't show up.

I think you are talking about this issue that was raised here https://github.com/hadynz/obsidian-kindle-plugin/issues/141. If so, then it should be implemented soon as recent changes to the plugin makes this a very easy feature to add

> What's the best way to start with? what's the biggest problem now?

That's a great question. I think part of this work is actually doing a discovery and understanding what are the problems with the current user interface. The user interface right now seems to do the job, but I wouldn't call it intuitive or insightful. It was hacked together to get a MVP out. 

We have an opportunity to make it much better. But that would require us to understand what problems do people syncing highlights want solved.

I know of at least one problem that was raised with the user interface that was raised by users: 
https://github.com/hadynz/obsidian-kindle-plugin/issues/150

**Reply by baipai** [reply]
@hadynz 
> I think you are talking about this issue that was raised here https://github.com/hadynz/obsidian-kindle-plugin/issues/141. If so, then it should be implemented soon as recent changes to the plugin makes this a very easy feature to add

Yes, I am talking about #141 

> That's a great question. I think part of this work is actually doing a discovery and understanding what are the problems with the current user interface. The user interface right now seems to do the job, but I wouldn't call it intuitive or insightful. It was hacked together to get a MVP out.

The discovery you are talking about here can mean different things, one is to discover the problem that already exists, another one is to discover other possibilities. What's your goal and plan?



> I know of at least one problem that was raised with the user interface that was raised by users: #150 

For #150, I also noticed that, and I use Amazon cloud. But hasn't the problem been solved? it seems there is a certain solution in the end? if not, what are you trying to learn more about?


> We have an opportunity to make it much better. But that would require us to understand what problems do people syncing highlights want solved.

When creating this plugin, do you have any specific group of users in mind? or do you know what kind of user is using this plugin more now? for what? and if there is anyone I can interview for? is the github the only place to let people discuss and get feedback?

Another thing is after checking all the existing github issues, I find out lots of them is because users have different habits when they create highlights, notes and also have different ways to use those notes after sync to Obsidian. #141 is one of them.


I think here are few actions I am going to do next: 
- Try the My clippings way to discover any problems
- Collect/group all the github issues and see if I can be able to find any more pattern inside
- Try to find people for interview or discussion from the community
- Check other service for Kindle notes to learn more the possible user group and some other possibilities 

Let me know what you think of this.

**Reply by hadynz** [reply]
> I think here are few actions I am going to do next: 
> Try the My clippings way to discover any problems
> Collect/group all the github issues and see if I can be able to find any more pattern inside
> Try to find people for interview or discussion from the community
> Check other service for Kindle notes to learn more the possible user group and some other possibilities 

The steps that you've just mentioned above make a LOT of sense and are a reasonable place to start. Theming digital feedback about the plugin and interviewing people would immediately help us understand what unsolved problems are out there that we can start focusing on.

If you are still keen to help and have the capacity, let's lock this in. We can look at another medium to kick off the collaboration and plan how we work together.

**Reply by baipai** [reply]
@hadynz 
Yeah, sure, I am keen on this, this is very interesting. How about let's find a time to meet and talk about how to work together? My email is bai@paiyuhsuan.com.
